.travis.yml

```diff
@@ -1,46 +1,16 @@
 language: cpp
 os: linux
-sudo: required
+dist: xenial
 services: docker
 
-branches:
-  only:
-  - master
-  - devel
-  # regex to match valid tag format or TravisCI won't trigger deploys when a new tag is pushed
-  - /^v[0-9]+\.[0-9]+\.[0-9]+.*$/
+addons:
+  apt:
+    packages:
+      - libboost-system-dev
+      - libmysqlclient-dev
+      - socat
 
 script:
-- make CONCURRENCY=4 package-debian-buster-dbuild
-- make CONCURRENCY=4 package-debian-stretch-dbuild
-- make CONCURRENCY=4 package-debian-jessie-dbuild
-- make CONCURRENCY=4 package-fedora-22-dbuild
-
-before_deploy: |
-  # move packages to ARTIFACTS directory
-  pushd packages
-  mkdir ARTIFACTS
-  ## rename debian packages to be distribution specific and move them to ARTIFACTS directory
-  for DEBIAN_PKG in debian-*/libsuperiormysqlpp-dev_*.deb; do
-    GITHUB_ARTIFACT_NAME=`echo ${DEBIAN_PKG} | sed --regexp-extended 's|debian-([^/]+)/libsuperiormysqlpp-dev_([^_]+)_amd64.deb|ARTIFACTS/libsuperiormysqlpp-dev_\2-\1_amd64.deb|'`
-    echo "Renaming & moving package '${DEBIAN_PKG}' to '${GITHUB_ARTIFACT_NAME}'"
-    mv ${DEBIAN_PKG} ${GITHUB_ARTIFACT_NAME}
-  done
-  ## move RPM packages to ARTIFACTS directory
-  mv --target-directory=ARTIFACTS fedora-*/*.rpm
-  popd
-
-  # GitHub releaser script (https://github.com/lindell/github-release-cli)
-  # used because standard github-release deploy module doesn't support specifying release message from ENV
-  curl https://github.com/lindell/github-release-cli/releases/download/1.3.0/github-releaser-travis --fail --location --output github-releaser && chmod +x github-releaser \
-    || (echo "Failed to download github-releaser script" && return 1)
-  ## get changes from debian's changelog
-  export BODY=`dpkg-parsechangelog --file packages/_debian-common/changelog --show-field Changes | grep --invert-match --extended-regexp '^$|^libsuperiormysqlpp |^  \[ '`
-  export FILES=packages/ARTIFACTS/*
-
-deploy:
-  provider: script
-  script: ./github-releaser -draft -verbose
-  skip_cleanup: true
-  on:
-    tags: true
+  - mkdir build && cd build && cmake .. -DTEST_ENABLE=TRUE
+  - cmake --build . -j 6
```

Comment:
I had used this with Unix Makefiles, Visual Studio, and possibly Ninja and basically, I did never had problems with building for different platforms and never needed to specify platform dependent building commands.
CMake can do basically both builds, but that depends on what it'll generate. It kinda does both things. When Cmake is called in dedicated directory it'll always do out-source and keeps root folder nicely clean from all sorts of trash.
I kinda despise in-source builds, since they clutter all sorts of files in a root folder anyway.